app-api-common/recipe/src/main/java/org/codeforvictory/android/recipes/data/api/recipe/RecipeServiceProvider.kt

```diff
@@ -0,0 +1,17 @@
+package org.codeforvictory.android.recipes.data.api.recipe
+
+import retrofit2.Retrofit
+
+internal class RecipeServiceProvider(private val retrofit: Retrofit = RetrofitProvider.value) {
+
+    private var singleton: RecipeService? = null
+    val value: RecipeService
+        get() {
+            if (singleton == null) {
+                synchronized(this) {
+                    singleton = retrofit.create(RecipeService::class.java)
```

Comment:
here we another check inside the synchronized, otherwise we will be create the singleton multiple times:
https://www.baeldung.com/java-singleton-double-checked-locking

Reply:
Addressed: [`4606ac9` (#13)](https://github.com/CodeForVictory/recipes-mvi-playground/pull/13/commits/4606ac9ce1e6d7155cba6ed2f1f3145ed6ee3929)